web_services/foxx/brightid/db.js

```diff
@@ -567,9 +575,17 @@ function getLastContextIds(coll, appKey) {
 }
 
 function userVerifications(user) {
-  const verifications = verificationsColl.byExample({
-    user
-  }).toArray();
+  const hashes = variablesColl.document('VERIFICATIONS_HASHES').hashes;
+  const snapshotPeriod = hashes[1]['block'] - hashes[0]['block']
+  const lastBlock = variablesColl.document('LAST_BLOCK').value;
+  // We want verifications from the second-most recently finished snapshot
+  // prior to LAST_BLOCK. For example if
+  //   last last processed block by consensus service (lastBlock) is 520
+  //   and we create and process a new snapshot (snapshotPeriod) every 100 blocks
+  // we want verifications from block 400 and not 500, to ensure all nodes
+  // had enough time (at least equal time to one snapshot period) to calculate them.
+  const block = lastBlock - (lastBlock % snapshotPeriod) - snapshotPeriod
```

Comment:
Ok so if I understand correctly, if the node is far behind, it's going to look in the db for verifications it doesn't have yet
and then return an empty array of verifications
even if the user is really verified
But you said this should never happen I think.
@abramsymons 
>such a thing should not ever happen. Snapshot period should be long enough to ensure all synced nodes have same understanding on verifications on consensus block.

So we'll just accept this code as-is and say that this never happens.